services/mpa/client/client.go

```diff
@@ -113,20 +116,47 @@ func (*approveCmd) Usage() string {
 `
 }
 
-func (p *approveCmd) SetFlags(f *flag.FlagSet) {}
+func (p *approveCmd) SetFlags(f *flag.FlagSet) {
+	f.BoolVar(&p.skipConfirmation, "skip-confirmation", false, "If true won't ask for confirmation")
+}
 
 func (p *approveCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
 	state := args[0].(*util.ExecuteState)
 	if f.NArg() != 1 {
 		fmt.Fprintln(os.Stderr, "Please specify a single ID to approve.")
 		return subcommands.ExitUsageError
 	}
+	id := f.Args()[0]
 	c := pb.NewMpaClientProxy(state.Conn)
-	action := getAction(ctx, state, c, f.Args()[0])
+	action := getAction(ctx, state, c, id)
 	if action == nil {
 		return subcommands.ExitFailure
 	}
 
+	if !p.skipConfirmation {
+		// ask for confirmation
+		fmt.Printf("MPA Request:\n%s\n", protojson.MarshalOptions{UseProtoNames: true, Multiline: true}.Format(action))
```

Comment:
I forgot that we were already fetching this. What do you think of unconditionally printing the request?

Reply:
I really like that idea.

Reply:
Changed to always print the request